### PR TITLE
Implemented Step 1 of bp:logical-read-ahead (Merge aio page read requests) (5.7)

### PR DESCRIPTION
**Pull request by percona-ysorokin**

### CODE REVIEW

File: storage/innobase/os/os0file.cc [hunk 1]
```diff
@@ -6547,6 +6666,13 @@ AIO::init_linux_native_aio()
 		}
 	}
 
+	m_pending = static_cast<struct iocb**>(
+		ut_malloc_nokey(m_slots.size() * sizeof(struct iocb*)));
```
Comment (laurynas-biveinis) [inline storage/innobase/os/os0file.cc]: ut_zalloc_nokey ?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: storage/innobase/os/os0file.cc (hunk 1 above)
Comment (percona-ysorokin) [reply storage/innobase/os/os0file.cc]: fixed

**Comment by percona-ysorokin** [comment]
Jenkins build link
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1201

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by laurynas-biveinis** [comment]
@percona-ysorokin, this needs a separate 5.7 blueprint on Launchpad

**Comment by percona-ysorokin** [comment]
bp:merge-aio-page-read-requests-5.7 registered
https://blueprints.launchpad.net/percona-server/+spec/merge-aio-page-read-requests-5.7

**Pull request merged**